File: 4.ROS/Tasks/Task_4/nomeer_ws/src/cmplx_num_pkg/src/sub_cmplx.cpp
#include <ros/ros.h>
#include <cmplx_num_pkg/cmplx.h>


void callback(const cmplx_num_pkg::cmplx::ConstPtr& cmplxNum)
{
    ROS_INFO("(%.3fi + %.3fj)", cmplxNum->real, cmplxNum->imaginary);

}
int main(int argc, char** argv)
{
    ros::init(argc, argv, "sub_cmplx_cpp");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("cmplx_topic_cpp",1,callback);
    ros::spin();

    return 0;
}

File: Task_4/nomeer_ws/src/cmplx_num_pkg/scripts/sub_cmplx.py
#!/usr/bin/python3

import rospy
from cmplx_num_pkg.msg import cmplx


def callback(cmplx):
    rospy.loginfo("(%.3fi + %.3fj)", cmplx.real, cmplx.imaginary)
def sub_cmplx():
    rospy.init_node("sub_cmplx")
    sub = rospy.Subscriber("cmplx_topic", cmplx, callback)
    rospy.spin()


if __name__ == "__main__":
    sub_cmplx()